pyproject.toml

```diff
@@ -42,6 +43,7 @@ select = [
     "I",
     "ISC",
     "D",
+    "UP"
```

Comment:
I'd appreciate if we could introduce this rule in a separate commit from the ones bumping the supported Python version window (possibly as a separate commit that's before the rest of the logic).

Not opposed to adding it, just making the changes slightly more atomic.

Reply:
This is added with #230, will remove from this PR.